lib/action_view/component/base.rb

```diff
@@ -171,6 +173,14 @@ def compile(validate: false)
             return false
           end
 
+          if self == PublicMethodComponent 
```

Comment:
I don't understand this check.

Reply:
This is what I get for not opening this is a draft 🤦‍♂ . This is still a work in progress! I'll close it for now.